### PR TITLE
Fix crash on commission rates input

### PR DESCRIPTION
**Pull request by Radiokot**
## Purpose

Fix inability to edit validator pool commission rates in locales with comma decimal separator due to crashing.

## Changes

- Simplify setting of input suffix by making it a companion `TextView` – `TextWatcher` approach doesn't go well with decimal input
- Make rate inputs `AmountEditText`s to properly handle decimal input
- Do not rewrite input values on failed parsing
- Use locale-aware number format for percentages
- Remove obsolete extensions

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: app/src/main/res/layout/activity_baker_settings.xml
Comment (shjortConcordium) [inline app/src/main/res/layout/activity_baker_settings.xml:82]:
```suggestion
                        app:layout_constraintWidth="wrap_content"
                        app:layout_constraintWidth_min="10dp"
```
Can we add a min size for these Edit Views, so that if the amount is deleted, one can still click them?

File: app/src/main/res/layout/activity_baker_settings.xml
Comment (shjortConcordium) [inline app/src/main/res/layout/activity_baker_settings.xml:166]:
```suggestion
                        app:layout_constraintWidth="wrap_content"
                        app:layout_constraintWidth_min="10dp"
```
Same here :upside_down_face:

**Review by shjortConcordium — APPROVED** [review]
LGTM, please consider my suggestion.

File: app/src/main/res/layout/activity_baker_settings.xml
Comment (Radiokot) [reply app/src/main/res/layout/activity_baker_settings.xml:82]: Makes sense, I think I'll also make clicking the "%" focus the corresponding field.

Comment (Radiokot) [reply app/src/main/res/layout/activity_baker_settings.xml:82]: I'll add the suggestion in the release branch not to wait for another review.

**Pull request merged**
